src/text_edit.rs

```diff
@@ -132,13 +132,13 @@ pub fn apply_text_edits_to_buffer(
     text_edits: &[OneOf<TextEdit, AnnotatedTextEdit>],
     text: &Rope,
     offset_encoding: OffsetEncoding,
-) -> String {
+) -> Option<String> {
     // Empty text edits processed as a special case because Kakoune's `select` command
     // doesn't support empty arguments list.
     if text_edits.is_empty() {
         // Nothing to do, but sending command back to the editor is required to handle case when
         // editor is blocked waiting for response via fifo.
-        return "nop".to_string();
```

Comment:
I think fn shouldn't know what we suppose to send back to editor, thus just returning None.